File: dotnetBackEnd/Shared/PlayerDto.cs
using Domain;
using System;

namespace Shared
{
    public static class PlayerDto
    {
        public class Create
        {
            public int PlayerID { get; set; }
            public string Name { get; set; }
            public bool Gender { get; set; } //true for male, false for female
            public bool Active { get; set; }
            public int? Birthyear { get; set; }
            public string PlayHand { get; set; }
            public string PlayStyle { get; set; }
            public string Grip { get; set; }
            public string CountryCode { get; set; }
        }

        public class Delete
        {
            public Guid Id { get; set; }
        }

        public class Update
        {
            public Guid Id { get; set; }
            public string Name { get; set; }
            public bool Active { get; set; }
        }
    }
}

File: vaporBackEnd/Sources/App/DTO/PlayerDTO.swift
//
//  File.swift
//  
//
//  Created by Benny De Bock on 26/05/2022.
//

import Foundation
import Vapor

struct PlayerDTO {
    struct GetById: Content {
        let id: UUID?
    }
    struct Delete: Content {
        let Id: UUID
    }
    
    struct Create: Content {
        let PlayerID: Int
        let Name: String
        let Gender: Bool
        let Active: Bool
        let Birthyear: Int?
        let PlayHand: String?
        let PlayStyle: String?
        let Grip: String?
        let CountryCode: String
    }
    
    struct Update: Content {
        let Id: UUID
        let Name: String
        let Active: Bool
    }
}
